pkg/resource/compare/defaults.go

```diff
@@ -199,6 +176,107 @@ func equalDeploymentConfigs(deployed resource.KubernetesResource, requested reso
 	return equal
 }
 
+func equalDeployment(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
+	d1 := deployed.(*appsv1.Deployment)
+	d2 := requested.(*appsv1.Deployment)
+
+	d1 = d1.DeepCopy()
+	triggerBasedImage := make(map[string]bool)
+
+	if d2.Spec.Strategy.RollingUpdate == nil && d1.Spec.Strategy.RollingUpdate != nil {
+		d1.Spec.Strategy.RollingUpdate = nil
+	}
+	if d1.Spec.Strategy.RollingUpdate != nil && d2.Spec.Strategy.RollingUpdate != nil {
+		if d2.Spec.Strategy.RollingUpdate.MaxSurge == nil {
+			d1.Spec.Strategy.RollingUpdate.MaxSurge = nil
+		}
+		if d2.Spec.Strategy.RollingUpdate.MaxUnavailable == nil {
+			d1.Spec.Strategy.RollingUpdate.MaxUnavailable = nil
+		}
+	}
+	if d2.Spec.RevisionHistoryLimit == nil {
+		d1.Spec.RevisionHistoryLimit = nil
+	}
+	if d2.Spec.ProgressDeadlineSeconds == nil {
+		d1.Spec.ProgressDeadlineSeconds = nil
+	}
+
+	if &d1.Spec.Template != nil && &d2.Spec.Template != nil {
+		for k, v := range d1.Annotations {
```

Comment:
Instead of testing if a key contains `imageTriggersAnnotation`, we can do an assignment tests to check if the annotation exists:
```suggestion
		if value, ok := d1.Annotations[imageTriggersAnnotation]; ok {
		   ...
		}
```